src/data/set/basic.lean

```diff
@@ -2936,3 +2947,46 @@ instance decidable_set_of (p : α → Prop) [decidable (p a)] : decidable (a ∈
 by assumption
 
 end set
+
+/-! ### Indicator function valued in bool -/
+
+open bool
+
+namespace set
+variables {α : Type*} (s : set α)
+
+/-- `bool_indicator` maps `x` to `tt` if `x ∈ s`, else to `ff` -/
+noncomputable def bool_indicator (x : α) :=
+@ite _ (x ∈ s) (classical.prop_decidable _) tt ff
```

Comment:
Once you generalise according to my previous comment, you won't need this anymore.

Reply:
Because `bool` is the simplest non-preconnected space, having an indicator function valued in `bool` is convenient. I thought about making a local `has_zero bool` instance and define `bool_indicator` in terms of `set.indicator`, or at least deduce lemmas from the `set.indicator` counterparts to reduces duplication, but it seems not many of them can be easily deduced, and the API for `bool_indicator` is sufficiently distinct from `set.indicator` to deserve a separate declaration. So I'm satisfied with the current situation.